packages/block-editor/src/utils/object.js

```diff
@@ -27,7 +27,7 @@ function normalizePath( path ) {
  * @return {*} Cloned object, or original literal non-object value.
  */
 function cloneObject( object ) {
-	if ( typeof object === 'object' ) {
+	if ( object && typeof object === 'object' ) {
```

Comment:
Is there a test case that can be added to the tests for `immutableSet`?

BTW, `cloneObject` introduced in https://github.com/WordPress/gutenberg/pull/49365, it was already in the codebase. I think it might be from the refactor away from lodash, so it might be good to check what lodash does in this situation and mirror that.

Reply:
It looks like this is what happens:
```js
_.set( null, [ 'path' ], 'value' ); // returns `null`
_.set( undefined, [ 'path' ], 'value' ); // returns `undefined`
```

So I think the fix is good in that it still returns the original value.